src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java

```diff
@@ -164,23 +165,51 @@ private void formaliseResponseAcceptance(
         }
     }
 
-    private void raiseAppInsightEvents(Claim claim, Response response, ClaimantResponse claimantResponse) {
+    private void raiseAppInsightEvents(Claim claim,
+                                        Response response,
+                                        ClaimantResponse claimantResponse,
+                                        Optional<CaseEvent> caseEvent) {
+
         if (claimantResponse instanceof ResponseAcceptation) {
             appInsights.trackEvent(CLAIMANT_RESPONSE_ACCEPTED, REFERENCE_NUMBER, claim.getReferenceNumber());
+
         } else if (claimantResponse instanceof ResponseRejection) {
             if (isPartAdmissionOrIsStatePaidOrIsFullDefence(response)) {
-                raiseAppInsightEventForLegalAdvisorPilot(claim);
+                raiseAppInsightEventForOnlineOrOfflineDQ(claim);
                 raiseAppInsightEventForMediation(claim, response, (ResponseRejection) claimantResponse);
+                raiseAppInsightsEventForPilot(claim, caseEvent);
             }
         } else {
             throw new IllegalStateException("Unknown response type");
         }
     }
 
-    private void raiseAppInsightEventForLegalAdvisorPilot(Claim claim) {
-        AppInsightsEvent appInsightsEvent = FeaturesUtils.isLegalAdvisorPilot(claim)
-            ? AppInsightsEvent.LA_PILOT_ELIGIBLE
-            : AppInsightsEvent.NON_LA_CASES;
+    private void raiseAppInsightsEventForPilot(Claim claim, Optional<CaseEvent> caseEvent) {
```

Comment:
Prefer not to pass `Optional` around as parameters

Reply:
passing Optional to method param is bad practise